src/main/java/Duke.java

```diff
@@ -1,10 +1,240 @@
+                        task_list.add(todo);
+                        PrintTask(task_list, todo);
+                        break;
+
+                    case "deadline":
+                        String date_n_time_d[] = task_details.split("by ")[1].split(" ");
+                        String str_date_d = date_n_time_d[0];
+                        LocalDate date_d = LocalDate.parse(str_date_d);
+                        String time_d = date_n_time_d[1];
+                        Task deadline = new Deadline(task_info, date_d, time_d);
+                        task_list.add(deadline);
+                        PrintTask(task_list, deadline);
+                        break;
+
+                    case "event":
+                        String date_n_time_e[] = task_details.split("at ")[1].split(" ");
+                        String str_date_e = date_n_time_e[0];
+                        LocalDate date_e = LocalDate.parse(str_date_e);
+                        String time_e = date_n_time_e[1];
+                        Task event = new Event(task_info, date_e, time_e);
+                        task_list.add(event);
+                        PrintTask(task_list, event);
+                        break;
+
+                    case "done":
+                        Task temp = task_list.get(Integer.parseInt(task_details) - 1);
+                        temp.completeTask();
+                        System.out.println("    ____________________________________________________________\n" +
+                                "     Nice! I've marked this task as done: \n" +
+                                "       " + temp + "\n" +
+                                "    ____________________________________________________________");
+                        break;
+
+                    case "delete":
+                        Task temp2 = task_list.remove(Integer.parseInt(task_details) - 1);
+                        System.out.println(
+                                String.format("    ____________________________________________________________\n" +
+                                "     Noted. I've removed this task: \n" +
+                                "       %s\n" +
+                                "     Now you have %d tasks in the list.\n" +
+                                "    ____________________________________________________________", temp2.toString(), task_list.size()));
+
+                    default:
+                        break;
+                }
+            }
+            catch (ArrayIndexOutOfBoundsException e) {
+                throw new DukeIncompleteCommandException(String.format(
+                        "    ____________________________________________________________\n" +
+                        "     ☹ OOPS!!! The task_info of a %s cannot be empty.\n" +
+                        "    ____________________________________________________________\n", task_type));
+            }
+    }
```

Comment:
Personally, I think this method is a bit too complicated, and it may get more complicated if you want to add more commands and features. So, I think maybe can consider about separating out some smaller methods from this long method, or separate some functions from this method to other classes.